bin/generate-docs/lib/file-system.js

```diff
@@ -145,25 +150,32 @@ function isFile( file ) {
 /**
  * Create a table of contents given a list of markdown files.
  *
- * @param { array } files A list of readme files.
+ * @param { array } files A list of files, presumably in the components directory.
+ * @param { string } route Folder where the docs are stored.
+ * @param { string } title Title of the TOC section
+ * @return { string } TOC contents.
  */
-function writeTableOfContents( files ) {
-	const mdFiles = files.map( f => getMdFileName( f, false ) ).sort();
-	const TOC = uniq( mdFiles ).map( doc => {
+function getTocContents( files, route, title ) {
```

Comment:
I like this, splitting up the generation of the TOC and the writing of the file.